test/00_transform/ResamplePoly.cu

```diff
@@ -0,0 +1,433 @@
+////////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// Copyright (c) 2021, NVIDIA Corporation
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this
+//    list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice,
+//    this list of conditions and the following disclaimer in the documentation
+//    and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its
+//    contributors may be used to endorse or promote products derived from
+//    this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+/////////////////////////////////////////////////////////////////////////////////
+
+#include "assert.h"
+#include "matx.h"
+#include "test_types.h"
+#include "utilities.h"
+#include "gtest/gtest.h"
+
+using namespace matx;
+
+template <typename T>
+class ResamplePolyTest : public ::testing::Test {
+protected:
+  void SetUp() override
+  {
+    CheckTestTypeSupport<T>();
+    pb = std::make_unique<detail::MatXPybind>();
+
+    if constexpr (is_complex_half_v<T> || is_matx_half_v<T>) {
+      thresh = 0.2;
+    } else if constexpr (std::is_same_v<T, double>) {
+      thresh = 1.0e-12;
+    } else {
+      // Revisit this tolerance. We should likely use a relative tolerance
+      // rather than absolute for larger values.
+      thresh = 1.0e-1;
+    }
+  }
+
+  void TearDown() { pb.reset(); }
+
+  std::unique_ptr<detail::MatXPybind> pb;
+  double thresh;;
+};
+
+template <typename TensorType>
+class ResamplePolyTestFloatTypes
+    : public ResamplePolyTest<TensorType> {
+};
+
+TYPED_TEST_SUITE(ResamplePolyTestFloatTypes, MatXFloatNonHalfTypes);
```

Comment:
Is there a reason half doesn't work? Could it be tested with a large precision?

Reply:
I added some test cases with short filters and enabled half types for those.